aten/src/ATen/native/quantized/cpu/qnnpack/test/gemm-block-sparse-microkernel-tester.h

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <cstdlib>
+#include <functional>
+#include <random>
+#include <vector>
+
+#include <fp16.h>
+
+#include <qnnpack/AlignedAllocator.h>
+#include <qnnpack/pack_block_sparse.h>
+#include <qnnpack/params.h>
+#include <qnnpack/requantization.h>
+
+namespace {
```

Comment:
This doesn't matter since this is a unit test but placing anonymous or static functions in a header file means each compilation unit is going to end up with its own copy since this implies internal linkage.  I am not actually sure if the compiler consolidates all [similar] functions into one in practice but avoid this usage personally.

Reply:
I think you are right. It will just create a separate copy for each translation unit. But I am more lax with test code and did this just out of practice.